inc/Engine/Preload/Controller/CheckExcludedTrait.php

```diff
@@ -14,4 +14,30 @@ protected function is_excluded( string $url ) {
 		$excluded = str_replace( '/', '\/', get_rocket_cache_reject_uri() );
 		return (bool) preg_match( "/$excluded/", $url );
 	}
+
+	/**
+	 * Check if the url is excluded by using a filter.
+	 *
+	 * @param string $url url to check.
+	 * @return bool
+	 */
+	protected function is_excluded_by_filter( string $url ) {
+		/**
+		 * Regex to exclude URI from preload.
+		 *
+		 * @param string[] regexes to check
+		 */
+		$regexes = apply_filters( 'rocket_preload_exclude_urls', [] );
+
+		if ( empty( $regexes ) ) {
+			return false;
+		}
+
```

Comment:
U need casting here and make sure that this array contains strings only
```suggestion
		$regexes = (array) apply_filters( 'rocket_preload_exclude_urls', [] );

		if ( empty( $regexes ) ) {
			return false;
		}
		
		$regexes = array_filter( $items, 'is_string' );
```